platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java

```diff
@@ -891,6 +909,16 @@ private ImportResponse configureProject(ImportProject importProject, FolderEntry
         return importResponse;
     }
 
+    private boolean resolveProjectType(FolderEntry projectFolder) {
```

Comment:
I think that we already have the similar functionality there:
ProjectManager#estimateProject()
ProjectManager#resolveSources()